lib/protobuf/enum.rb

```diff
@@ -198,7 +206,7 @@ def self.name_for_tag(tag)
     # Returns a boolean.
     #
     def self.valid_tag?(tag)
-      tag.respond_to?(:to_i) && all_tags.include?(tag.to_i)
+      tag.respond_to?(:to_i) && mapped_enums.key?(tag.to_i)
```

Comment:
is `respond_to?` usually expected to be `true`? If yes, maybe catching `NoMethodError` is preferable - should be faster in typical case, slower in error case.